monkey/monkey_island/cc/ui/src/components/attack/MatrixComponent.js

```diff
@@ -0,0 +1,196 @@
+import React from 'react';
+import Checkbox from '../ui-components/checkbox'
+import Tooltip from 'react-tooltip-lite'
+import AuthComponent from '../AuthComponent';
+import ReactTable from "react-table";
+import 'filepond/dist/filepond.min.css';
+import '../../styles/Tooltip.scss';
+
+
+// Finds which attack type has most techniques and returns that number
+let findMaxTechniques =  function (data){
+  let maxLen = 0;
+  data.forEach(function(techType) {
+    if (Object.keys(techType.properties).length > maxLen){
+      maxLen = Object.keys(techType.properties).length
+    }
+  });
+  return maxLen
+};
+
+// Parses ATT&CK config schema into data suitable for react-table (ATT&CK matrix)
+let parseTechniques = function (data, maxLen) {
+  let techniques = [];
+  // Create rows with attack techniques
+  for (let i = 0; i < maxLen; i++) {
+    let row = {};
+    data.forEach(function(techType){
+      let rowColumn = {};
+      rowColumn.techName = techType.title;
+
+      if (i <= Object.keys(techType.properties).length) {
+        rowColumn.technique = Object.values(techType.properties)[i];
+        if (rowColumn.technique){
+          rowColumn.technique.name = Object.keys(techType.properties)[i]
+        }
+      } else {
+        rowColumn.technique = null
+      }
+      row[rowColumn.techName] = rowColumn
+    });
+    techniques.push(row)
+  }
+  return techniques;
+};
+
+class MatrixComponent extends AuthComponent {
+  constructor(props) {
+    super(props);
+    // Copy ATT&CK configuration and parse it for ATT&CK matrix table
+    let configCopy = JSON.parse(JSON.stringify(this.props.configuration));
+    this.state = {lastAction: 'none',
+                  configData: this.props.configuration,
+                  maxTechniques: findMaxTechniques(Object.values(configCopy))};
+    this.state.matrixTableData = parseTechniques(Object.values(configCopy), this.state.maxTechniques);
+    this.state.columns = this.getColumns(this.state.matrixTableData)
+  };
+
+  getColumns(matrixData) {
+    return Object.keys(matrixData[0]).map((key)=>{
+      return {
+        Header: key,
+        id: key,
+        accessor: x => this.renderTechnique(x[key].technique),
+        style: { 'whiteSpace': 'unset' }
+      };
+    });
+  }
+
+  renderTechnique(technique) {
+    if (technique == null){
+      return (<div></div>)
+    } else {
+      return (<Tooltip content={technique.description} direction="down">
```

Comment:
Couldn't find a way to render `content` as HTML. I think it would be better if we could use HTML in tooltip, not only plain string. But, since that HTML comes from back end, react renders it as plain string.

Reply:
@itaymmguardicore